src/Compilers/CSharp/Portable/Utilities/ValueSetFactory.NumericValueSet.cs

```diff
@@ -49,7 +51,23 @@ internal NumericValueSet(ImmutableArray<(T first, T last)> intervals)
                 _intervals = intervals;
             }
 
-            bool IValueSet.IsEmpty => _intervals.Length == 0;
+            public bool IsEmpty => _intervals.Length == 0;
+
+            ConstantValue IValueSet.Sample
+            {
+                get
+                {
+                    // Prefer a value near zero.
+                    var tc = default(TTC);
+                    var gz = NumericValueSetFactory<T, TTC>.Instance.Related(BinaryOperatorKind.GreaterThanOrEqual, tc.Zero);
+                    var t = (NumericValueSet<T, TTC>)this.Intersect(gz);
+                    if (!t.IsEmpty)
+                        return tc.ToConstantValue(t._intervals[0].first);
+                    t = (NumericValueSet<T, TTC>)this.Intersect(gz.Complement());
+                    Debug.Assert(!t.IsEmpty);
```

Comment:
It feels like we should throw an exception instead of asserting to meet the requirement that this property throws when the set is empty.